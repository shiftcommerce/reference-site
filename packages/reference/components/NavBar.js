import { Component } from 'react'

// Objects
import NavBarOption from './NavBarOption'

class NavBar extends Component {
  renderNavOptions () {
    const navOptions = [
      { title: 'Mens', link: '/mens' },
      { title: 'Womens', link: '/womens' },
      { title: 'Boys', link: '/boys' },
      { title: 'Girls', link: '/girls' },
      { title: 'Other', link: '/other' },
      { title: 'Offers', link: '/offers' }
    ]

    return navOptions.map((navOption, index) =>
      <NavBarOption key={ index } index={ index } title={ navOption.title } link={ navOption.link } />
    )
  }

  render() {
    return <div className='o-nav' role='navigation' >
      { this.renderNavOptions() }
    </div>
  }
}

export default NavBar