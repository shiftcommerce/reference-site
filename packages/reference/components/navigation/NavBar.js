// Libraries
import { Component } from 'react'
import { connect } from 'react-redux'

// Objects
import NavBarOption from './NavBarOption'

// Actions
import { readCategories } from '../../actions/categoryActions'

class NavBar extends Component {
  componentDidMount () {
    this.props.dispatch(readCategories())
  }

  componentDidCatch (error, info) {
    console.log(error)
  }

  renderNavOptions (categories) {
    return (
      categories.data.map((category, index) =>
        <NavBarOption key={index} index={index} title={category.title} href={`/categories/category?id=${category.id}`} as={`/categories/${category.id}`} />
      )
    )
  }

  render () {
    const {
      categories
    } = this.props

    const hasData = categories.data && categories.data.length > 0

    return (
      <div className='o-nav' role='navigation' >
        { hasData && this.renderNavOptions(categories) }
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    categories: state.categories || []
  }
}

export default connect(mapStateToProps)(NavBar)
