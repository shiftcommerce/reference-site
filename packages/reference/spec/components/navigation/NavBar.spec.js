// Libraries
import { createMockStore } from 'redux-test-utils'

// Components
import NavBar from '../../../components/navigation/NavBar'

// Fixtures
import menu from '../../fixtures/menu.fixture'

test('renders the nav bar options', () => {
  // Arrange
  const initialState = {
    menu: {
      loading: false,
      error: false,
      data: menu
    }
  }

  const context = {
    context: {
      store: createMockStore(initialState)
    }
  }

  // Act
  const wrapper = mount(<NavBar />, context)

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toHaveHTML('<div class="o-nav" role="navigation"><a class="o-nav__option o-nav__option--first" href="/categories/1">Bathroom</a></div>')
})
