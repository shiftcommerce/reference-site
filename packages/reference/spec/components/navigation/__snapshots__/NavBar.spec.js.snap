// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the nav bar options 1`] = `
<Connect(NavBar)>
  <NavBar
    categories={
      Object {
        "data": Array [
          Object {
            "id": 1,
            "reference": "Bathroom",
            "title": "Bathroom",
          },
        ],
        "error": false,
        "loading": false,
      }
    }
    dispatch={[Function]}
  >
    <div
      className="o-nav"
      role="navigation"
    >
      <NavBarOption
        as="/categories/1"
        href="/categories/category?id=1"
        index={0}
        key="0"
        title="Bathroom"
      >
        <Link
          as="/categories/1"
          href="/categories/category?id=1"
        >
          <a
            className="o-nav__option o-nav__option--first"
            href="/categories/1"
            onClick={[Function]}
          >
            Bathroom
          </a>
        </Link>
      </NavBarOption>
    </div>
  </NavBar>
</Connect(NavBar)>
`;
